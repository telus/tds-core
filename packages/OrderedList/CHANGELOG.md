# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [3.0.20](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.19...@tds/core-ordered-list@3.0.20) (2020-07-22)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.19](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.18...@tds/core-ordered-list@3.0.19) (2020-04-14)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.18](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.17...@tds/core-ordered-list@3.0.18) (2020-02-24)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.17](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.16...@tds/core-ordered-list@3.0.17) (2020-02-03)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.16](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.15...@tds/core-ordered-list@3.0.16) (2020-01-27)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.15](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.14...@tds/core-ordered-list@3.0.15) (2020-01-23)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.14](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.13...@tds/core-ordered-list@3.0.14) (2020-01-22)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.13](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.12...@tds/core-ordered-list@3.0.13) (2019-10-09)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.12](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.11...@tds/core-ordered-list@3.0.12) (2019-10-04)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.11](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.10...@tds/core-ordered-list@3.0.11) (2019-09-12)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.10](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.9...@tds/core-ordered-list@3.0.10) (2019-09-06)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.9](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.8...@tds/core-ordered-list@3.0.9) (2019-08-22)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.8](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.7...@tds/core-ordered-list@3.0.8) (2019-08-09)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.7](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.6...@tds/core-ordered-list@3.0.7) (2019-08-01)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.6](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.5...@tds/core-ordered-list@3.0.6) (2019-07-30)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.5](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.4...@tds/core-ordered-list@3.0.5) (2019-07-25)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.4](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.3...@tds/core-ordered-list@3.0.4) (2019-07-17)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.3](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.2...@tds/core-ordered-list@3.0.3) (2019-06-25)

**Note:** Version bump only for package @tds/core-ordered-list





## [3.0.2](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.1...@tds/core-ordered-list@3.0.2) (2019-06-04)

**Note:** Version bump only for package @tds/core-ordered-list

## [3.0.1](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@3.0.0...@tds/core-ordered-list@3.0.1) (2019-05-13)

**Note:** Version bump only for package @tds/core-ordered-list

# [3.0.0](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.20...@tds/core-ordered-list@3.0.0) (2019-05-07)

### Features

- **core-ordered-list:** upgrade to styled-components ([fe0323c](https://github.com/telusdigital/tds/commit/fe0323c))

### BREAKING CHANGES

- **core-ordered-list:** upgrade to styled-components

## [2.0.20](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.19...@tds/core-ordered-list@2.0.20) (2019-05-06)

### Bug Fixes

- **core-ordered-list:** use typography styles directly ([1a42884](https://github.com/telusdigital/tds/commit/1a42884))

## [2.0.19](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.18...@tds/core-ordered-list@2.0.19) (2019-05-01)

**Note:** Version bump only for package @tds/core-ordered-list

## [2.0.18](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.17...@tds/core-ordered-list@2.0.18) (2019-04-23)

**Note:** Version bump only for package @tds/core-ordered-list

## [2.0.17](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.16...@tds/core-ordered-list@2.0.17) (2019-04-22)

**Note:** Version bump only for package @tds/core-ordered-list

## [2.0.16](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.15...@tds/core-ordered-list@2.0.16) (2019-04-11)

### Bug Fixes

- **core-ordered-list:** use className prop for Box ([d998464](https://github.com/telusdigital/tds/commit/d998464))

## [2.0.15](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.14...@tds/core-ordered-list@2.0.15) (2019-04-08)

### Bug Fixes

- **core-ordered-list:** allowed conditional children rendering ([ab2ec94](https://github.com/telusdigital/tds/commit/ab2ec94))
- **core-ordered-list:** converted children to array before filtering ([2bca8a0](https://github.com/telusdigital/tds/commit/2bca8a0))

## [2.0.14](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.13...@tds/core-ordered-list@2.0.14) (2019-04-04)

**Note:** Version bump only for package @tds/core-ordered-list

## [2.0.13](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.12...@tds/core-ordered-list@2.0.13) (2019-03-28)

**Note:** Version bump only for package @tds/core-ordered-list

## [2.0.12](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.11...@tds/core-ordered-list@2.0.12) (2019-03-28)

**Note:** Version bump only for package @tds/core-ordered-list

## [2.0.11](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.10...@tds/core-ordered-list@2.0.11) (2019-03-20)

**Note:** Version bump only for package @tds/core-ordered-list

## [2.0.10](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.9...@tds/core-ordered-list@2.0.10) (2019-03-20)

**Note:** Version bump only for package @tds/core-ordered-list

## [2.0.9](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.8...@tds/core-ordered-list@2.0.9) (2019-03-18)

**Note:** Version bump only for package @tds/core-ordered-list

## [2.0.8](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.7...@tds/core-ordered-list@2.0.8) (2019-03-18)

**Note:** Version bump only for package @tds/core-ordered-list

## [2.0.7](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.6...@tds/core-ordered-list@2.0.7) (2019-03-11)

**Note:** Version bump only for package @tds/core-ordered-list

## [2.0.6](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.5...@tds/core-ordered-list@2.0.6) (2019-03-05)

**Note:** Version bump only for package @tds/core-ordered-list

## [2.0.5](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.4...@tds/core-ordered-list@2.0.5) (2019-02-06)

**Note:** Version bump only for package @tds/core-ordered-list

## [2.0.4](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.3...@tds/core-ordered-list@2.0.4) (2019-01-10)

**Note:** Version bump only for package @tds/core-ordered-list

<a name="2.0.3"></a>

## [2.0.3](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.2...@tds/core-ordered-list@2.0.3) (2018-12-19)

**Note:** Version bump only for package @tds/core-ordered-list

<a name="2.0.2"></a>

## [2.0.2](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.1...@tds/core-ordered-list@2.0.2) (2018-12-10)

**Note:** Version bump only for package @tds/core-ordered-list

<a name="2.0.1"></a>

## [2.0.1](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@2.0.0...@tds/core-ordered-list@2.0.1) (2018-12-06)

**Note:** Version bump only for package @tds/core-ordered-list

<a name="2.0.0"></a>

# [2.0.0](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@1.0.1...@tds/core-ordered-list@2.0.0) (2018-11-01)

### Bug Fixes

- **core-ordered-list:** adjust nested list spacing ([a9fccbc](https://github.com/telusdigital/tds/commit/a9fccbc))
- **core-ordered-list:** replace airbnb-prop-types with util-prop-types ([777c8cf](https://github.com/telusdigital/tds/commit/777c8cf))

### Features

- **core-ordered-list:** add [@tds](https://github.com/tds)/core-text as dependency ([72882d2](https://github.com/telusdigital/tds/commit/72882d2))
- **core-ordered-list:** handle nested list spacing, ordered list size ([8bfd3e5](https://github.com/telusdigital/tds/commit/8bfd3e5))
- **core-ordered-list:** update styles, add sizing prop ([9b80e6f](https://github.com/telusdigital/tds/commit/9b80e6f))

### BREAKING CHANGES

- **core-ordered-list:** add size prop to ordered list

<a name="1.0.1"></a>

## [1.0.1](https://github.com/telusdigital/tds/compare/@tds/core-ordered-list@1.0.0...@tds/core-ordered-list@1.0.1) (2018-04-02)

### Bug Fixes

- **core-box:** update tests ([c0f0343](https://github.com/telusdigital/tds/commit/c0f0343))
