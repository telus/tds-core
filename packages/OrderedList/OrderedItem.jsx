import React from 'react'
import PropTypes from 'prop-types'

import Text from '@tds/core-text'

import safeRest from '../../shared/utils/safeRest'

const OrderedItem = ({ children, size, ...rest }) => (
  <li {...safeRest(rest)}>
    <Text size={size}>{children}</Text>
  </li>
)

OrderedItem.propTypes = {
  children: PropTypes.node.isRequired,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
}

OrderedItem.defaultProps = {
  size: 'medium',
}

OrderedItem.displayName = 'OrderedList.Item'

export default OrderedItem
