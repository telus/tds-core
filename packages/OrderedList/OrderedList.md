```
  <OrderedList>
    <OrderedList.Item>
      Up to $400 credit available for new activations on select 2-year plans. Offer applies to new TELUS Business customer activations. Not available for Consumer accounts. Ask a TELUS representative for other eligibility requirements
    </OrderedList.Item>
    <OrderedList.Item>
      350 nationwide minutes share with other TELUS shareable mobility plans on the same account
    </OrderedList.Item>
    <OrderedList.Item>
      Unlimited nationwide minutes do not share
    </OrderedList.Item>
  </OrderedList>
```
