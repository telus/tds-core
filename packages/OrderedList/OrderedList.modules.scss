.decimal {
  composes: base from '../../shared/styles/List.modules.scss';
  list-style-type: decimal;
}

.upperAlpha {
  composes: base from '../../shared/styles/List.modules.scss';
  list-style-type: upper-alpha;
}

.lowerAlpha {
  composes: base from '../../shared/styles/List.modules.scss';
  list-style-type: lower-alpha;
}
