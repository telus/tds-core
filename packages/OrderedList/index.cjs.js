const OrderedList = require('./dist/index.cjs')

module.exports = OrderedList
