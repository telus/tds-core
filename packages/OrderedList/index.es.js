export { default } from './dist/index.es'
export * from './dist/index.es'
