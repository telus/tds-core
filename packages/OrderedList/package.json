{
  "name": "@tds/core-ordered-list",
  "version": "1.0.1",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telusdigital/tds.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telusdigital/tds/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": ">=15",
    "react-dom": ">=15"
  },
  "dependencies": {
    "@tds/core-box": "^1.0.1",
    "@tds/core-text": "^1.0.2",
    "prop-types": "^15.5.10"
  },
  "devDependencies": {
    "airbnb-prop-types": "^2.8.1"
  }
}
