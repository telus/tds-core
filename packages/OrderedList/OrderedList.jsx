import React from 'react'
import PropTypes from 'prop-types'
import { childrenOfType } from 'airbnb-prop-types'

import Box from '@tds/core-box'

import safeRest from '../../shared/utils/safeRest'
import joinClassNames from '../../shared/utils/joinClassNames'

import OrderedItem from './OrderedItem'

import styles from './OrderedList.modules.scss'

const injectListStyle = (child, size) => React.cloneElement(child, { size })

/**
 * @version ./package.json
 */
const OrderedList = ({ listStyle, size, children, ...rest }) => {
  return (
    <Box
      {...safeRest(rest)}
      tag="ol"
      between={2}
      dangerouslyAddClassName={joinClassNames(styles[listStyle], styles[size])}
    >
      {React.Children.map(children, child => injectListStyle(child, size))}
    </Box>
  )
}

OrderedList.propTypes = {
  /**
   * The bullet style.
   */
  listStyle: PropTypes.oneOf(['decimal', 'upperAlpha', 'lowerAlpha']),
  /**
   * The size of the list's text.
   */
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  /**
   * The list items. Must be at least one `OrderedList.Item`.
   */
  children: childrenOfType(OrderedItem).isRequired,
}

OrderedList.defaultProps = {
  listStyle: 'decimal',
  size: 'medium',
}

OrderedList.Item = OrderedItem

export default OrderedList
