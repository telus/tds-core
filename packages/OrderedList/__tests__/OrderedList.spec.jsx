import React from 'react'
import { shallow, render } from 'enzyme'

import OrderedList from '../OrderedList'

describe('<OrderedList />', () => {
  const doRender = (overrides = {}) =>
    render(
      <OrderedList {...overrides}>
        <OrderedList.Item>Lorem ipsum</OrderedList.Item>
        <OrderedList.Item>Dolor sit amet</OrderedList.Item>
      </OrderedList>
    )

  const doShallow = (overrides = {}) =>
    shallow(
      <OrderedList {...overrides}>
        <OrderedList.Item>Lorem ipsum</OrderedList.Item>
        <OrderedList.Item>Dolor sit amet</OrderedList.Item>
      </OrderedList>
    )

  it('renders', () => {
    const orderedList = doRender()

    expect(orderedList).toMatchSnapshot()
  })

  it('OrderList renders an HTML ol tag', () => {
    const orderedList = doShallow()

    expect(orderedList.dive()).toHaveTagName('ol')
  })

  it('OrderList.Item renders an HTML li tag', () => {
    const orderedListItem = shallow(<OrderedList.Item>a list item</OrderedList.Item>)

    expect(orderedListItem).toHaveTagName('li')
  })

  it('can have a list style', () => {
    let orderedList = doShallow({ listStyle: undefined })
    expect(orderedList.dive()).toHaveClassName('decimal')

    orderedList = doShallow({ listStyle: 'upperAlpha' })
    expect(orderedList.dive()).toHaveClassName('upperAlpha')
  })

  it('passes additional attributes to ol element', () => {
    const orderedList = doShallow({ id: 'the-list', reversed: 'true' })

    expect(orderedList).toHaveProp('id', 'the-list')
    expect(orderedList).toHaveProp('reversed', 'true')
  })

  it('does not allow custom CSS', () => {
    const orderedList = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(orderedList).not.toHaveProp('className', 'my-custom-class')
    expect(orderedList).not.toHaveProp('style')
  })
})
