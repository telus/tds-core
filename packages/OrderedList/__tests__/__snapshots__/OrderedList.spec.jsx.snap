// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderedList /> renders 1`] = `
<ol
  class="TDS_Box-modules__betweenBottomMargin-2___31zX_ TDS_Box-modules__stack___33m4D decimal medium"
>
  <li
    class="base"
  >
    <span
      class="TDS_Typography-modules__medium___1DC1g TDS_Typography-modules__wordBreak___3dmWU TDS_Typography-modules__mediumFont___2ULml TDS_Typography-modules__color___1Jt_W"
    >
      Lorem ipsum
    </span>
  </li>
  <li
    class="base"
  >
    <span
      class="TDS_Typography-modules__medium___1DC1g TDS_Typography-modules__wordBreak___3dmWU TDS_Typography-modules__mediumFont___2ULml TDS_Typography-modules__color___1Jt_W"
    >
      Dolor sit amet
    </span>
  </li>
</ol>
`;
